src/onnx-mlir.cpp

```diff
@@ -68,7 +69,14 @@ int main(int argc, char *argv[]) {
   }
 
   // Create context after MLIRContextCLOptions are registered and parsed.
-  mlir::MLIRContext context;
+  mlir::MLIRContext context(mlir::MLIRContext::Threading::DISABLED);
+  std::unique_ptr<llvm::ThreadPoolInterface> threadPoolPtr;
```

Comment:
What's the program behavior if setThreadPool is not set? I guess that the compiler uses all the available threads. 
What's wrong with j=1? Can we specify that only one thread is used? Then comes the question: what the default value should be?
We should add these descriptions In the option definition.
By the way,  I am surprised to say `Threading::DISABLED`, not enabled.

Reply:
@imaihal please look into this. Ideally, we would like to get the best performance and fastest compiler (using only `-O3` with possibly the arch and the use of the NNPA).

Chatting with @chentong319 , MLIR is default multiple threads, so we want to preserve that. The way to ask for a single thread would be explicitly stating `-j 1`

Reply:
@AlexandreEichenberger @chentong319 Thanks for the comments.

I see. I updated to use 0 by default and use all the available processor for compilation. When setting `-j 1`, use single thread without setting thread pool. When `j > 1`, specified the number for threads are used.

Any other requests that I don't refrect?

Reply:
That seems to be correct. Do you ignore (or warning) if user gave a negative value?

Reply:
This sounds great!